File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        num =0
        mapp = {'I': 1, 
            'V': 5, 
            'X': 10, 
            'L': 50, 
            'C': 100, 
            'D': 500, 
            'M': 1000}
        for i in range(len(s)):
            if (i<=len(s)-2 and mapp[s[i]]<mapp[s[i+1]]):
                num-=mapp[s[i]]
            else:
                num+=mapp[s[i]]
        return num

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character,Integer> mp = new HashMap<>();
        int num = 0;
        mp.put('I',1);
        mp.put('V',5);
        mp.put('X',10);
        mp.put('L',50);
        mp.put('C',100);
        mp.put('D',500);
        mp.put('M',1000);
        int ptr = 0;
        for(int i=1;i<s.length();i++){
            if (mp.get(s.charAt(ptr))>=mp.get(s.charAt(i))){
                num+=mp.get(s.charAt(ptr));
            }
            else{
                num-=mp.get(s.charAt(ptr));
            }
            ptr+=1;
        }
        num += mp.get(s.charAt(ptr));
        return num;
    }
}